daemon/api_debug_seeding.go

```diff
@@ -0,0 +1,107 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package daemon
+
+import (
+	"time"
+
+	"github.com/snapcore/snapd/overlord/state"
+)
+
+type seedingInfo struct {
+	// true if the device is seeded (same as "seeded" in the state)
+	Seeded bool `json:"seeded,omitempty"`
+
+	// true if snap-preseed was applied
+	Preseeded bool `json:"preseeded,omitempty"`
+
+	// when snap-preseed was started
+	PreseedStartTime time.Time `json:"preseed-start-time,omitempty"`
+
+	// when snap-preseed work finished (i.e. before firstboot)
+	PreseededTime time.Time `json:"preseeded-time,omitempty"`
+
+	// when traditional seeding started (when not preseeding)
+	SeedStartTime time.Time `json:"seed-start-time,omitempty"`
+
+	// when seeding was started after first boot of preseeded image
+	SeedRestartTime time.Time `json:"seed-restart-time,omitempty"`
+
+	// when seeding finished
+	SeedTime time.Time `json:"seed-time,omitempty"`
+
+	// system-key created during preseeding (when snap-preseed was ran)
+	PreseedSystemKey interface{} `json:"preseed-system-key,omitempty"`
+
+	// system-key created on first boot of the preseeded image
+	RestartSystemKey interface{} `json:"restart-system-key,omitempty"`
+}
+
+func getSeedingInfo(st *state.State) Response {
+	var seeded, preseeded bool
+	err := st.Get("seeded", &seeded)
+	if err != nil && err != state.ErrNoState {
+		return BadRequest(err.Error())
+	}
+	if err = st.Get("preseeded", &preseeded); err != nil && err != state.ErrNoState {
+		return BadRequest(err.Error())
```

Comment:
same here and most likely the other places

Reply:
I just noticed that this is set on the task which will be pruned away and not the state yet. That will need adjustments in the other PR.

Reply:
how do you mean? I see this code:
```
			st.Set("preseeded", preseeded)
```

where `st` is the state. Is that wrong ?

Reply:
(directed at @pedronis comment, I will change BadRequest -> InternalError here)

Reply:
no, the new PR is correct, I was referring at what's happening on master atm (though now in the new PR we have two flags one on the tasks and one on the state, I'm not sure whether to care or not), anyway logically the 3rd PR needs to land first I suppose

Reply:
ah ok, yes I see the issue with the task and the state, I think it's fine to leave them both around for now and clean that up after we have landed everything currently in flight

Reply:
Yes, I saw that when making these changes; before this it was sufficient to keep a flag on the task because it was used only to drive the logic of this task. We now need the flag on state and I decided to keep both for the moment, but we can get rid of the task flag.